clearing_operations/scripts/NumberFeaturesTool.py

```diff
@@ -0,0 +1,282 @@
+# coding: utf-8
+'''
+------------------------------------------------------------------------------
+ Copyright 2017 Esri
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+------------------------------------------------------------------------------
+ ==================================================
+ NumberFeatures_new.py
+ --------------------------------------------------
+ requirements: ArcGIS X.X, Python 2.7 or Python 3.x
+ author: ArcGIS Solutions
+ contact: support@esri.com
+ company: Esri
+ ==================================================
+ description: 
+ Number Features Tool logic module. 
+ Supports ClearingOperationsTools.pyt
+ ==================================================
+ history:
+ 9/6/2017 - mf - original coding/transfer from NumberFeatures.py
+ ==================================================
+'''
+
+
+import os
+import sys
+import traceback
+import arcpy
+from . import Utilities
+
+class NumberFeatures(object):
+    '''
+    Number input features within a specified area.
+    '''
+    def __init__(self):
+        '''
+        Number Features constructor
+        '''
+        self.label = "Number Features"
+        self.description = "Number input point features within a selected area."
+
+    def getParameterInfo(self):
+        '''
+        Define parameter definitions
+        '''
+        input_area_features = arcpy.Parameter(name='input_area_features',
+                                              displayName='Input Area to Number',
+                                              direction='Input',
+                                              datatype='GPFeatureRecordSetLayer',
+                                              parameterType='Required',
+                                              enabled=True,
+                                              multiValue=False)
+        input_area_features.value = os.path.join(os.path.dirname(os.path.dirname(__file__)),
```

Comment:
@mfunk @topowright @BobBooth 

I had the same problem with FeatureSets in ArcMap with the new .pyt that you mentioned - I got around this by switching from a .lyr file to actual feature class (fortunately there was already one there - perhaps for this reason/purpose?)

Modified code:

```
        schemaLayer = os.path.normpath(os.path.join(os.path.dirname(__file__), r'../layers/featuresetsWebMerc.gdb/InputArea'))
        input_area_features.value = schemaLayer
```

Reply:
That would get around part of the problem. Unfortunately we use symbology and feature templates (primarily for default edit tool) that are stored in the .LYR also. Fortunately I think we have nice solid (meaning, not devised by myself) fix for the problem.